doc/langref/test_multidimensional_arrays.zig

```diff
@@ -19,6 +19,10 @@ test "multidimensional arrays" {
             }
         }
     }
+
+    // initialize a multidimensional array to zeros
+    const all_zero = std.mem.zeroes([2][2]u16);
```

Comment:
I personally don't think this standard library function should be used in the langref here. It would teach more about the core language semantics to use something like
```zig
const all_zero: [10][10]u16 = [1][10]u16{[1]u16{0} ** 10} ** 10;
```

Reply:
Also, `std.mem.zeroes` is typically a code smell -- its primary use case is in dealing with C APIs.

Reply:
Between here and discord I've learned about:
```
const all_zero: [10][10]u16 = [1][10]u16{[1]u16{0} ** 10} ** 10;
const all_zero: [10][10]u16 = .{.{0}**10}**10;
const all_zero: [10][10]u16 = @bitCast([_]u16{0} ** (10 * 10));
const all_zero = std.mem.zeroes([2][2]u16);
```

I think this one is the most similar to the single dimension example so I'll update to that:
```
const all_zero: [10][10]u16 = .{.{0}**10}**10;
```

It would be nice to not have to write the lengths twice like in the single dimension example:
```
const all_zero = [_]u16{0} ** 10;
```